### PR TITLE
Improve duplicate detection with accent folding and significant words

### PR DESCRIPTION
**Pull request by mateusdeitos**
## Summary

Enhance transaction duplicate detection to handle accented characters and reduce false positives from generic bank-statement boilerplate. Descriptions like "Farmácia São João" now correctly match "Farmacia Sao Joao", and duplicates sharing a meaningful word (e.g., "Amazon (fraldas Luca)" vs "Amazon") are flagged even when trigram similarity is low.

## Key Changes

- **Accent folding**: Added `foldAccents()` to normalize Unicode combining marks (NFD decomposition), so "São" → "Sao" and "ç" → "c"
- **Shared tokenizer**: Extracted `normalizedWords()` to ensure consistent lowercasing, accent-folding, and word splitting across all similarity checks
- **Significant word matching**: Introduced `isSignificantWord()` and `sharesSignificantWord()` to flag duplicates sharing meaningful words (≥4 runes, not purely numeric, not generic bank terms)
- **Generic word list**: Added `genericDescriptionWords` map filtering out common Portuguese bank-statement terms ("compra", "pagamento", "cartao", "pix", etc.) that are too noisy to signal duplicates alone
- **Combined similarity logic**: New `descriptionsAreSimilar()` function unifies the check: trigram similarity ≥ threshold **OR** shared significant word
- **Updated call sites**: Replaced direct `trigramSimilarity()` comparisons in `filterDuplicateMatches()` and `filterSettlementDuplicateMatches()` with `descriptionsAreSimilar()`
- **Frontend label update**: Simplified the duplicate description label to reflect the new dual-signal approach
- **Test coverage**: Added comprehensive tests for accent folding, the new similarity function, and edge cases (empty strings, generic words, short tokens, numeric-only tokens)
- **Dependency**: Added `golang.org/x/text` for Unicode normalization

## Implementation Details

- Accent folding uses `unicode.Mn` (Mark, Nonspacing) category to drop combining marks after NFD decomposition
- Word significance threshold (4 runes) balances catching real duplicates ("Zp *Sunrize" shares "Sunrize") while rejecting noise ("Zp Padaria" vs "Zp Acougue" share only "Zp")
- Generic word list is pre-normalized (lowercase, accent-folded) to match `normalizedWords()` output directly
- Trigram padding and sliding window logic preserved; now operates on normalized words instead of the raw lowercased string

https://claude.ai/code/session_019e5Npvx11qTrSquTPAmJ34

### CODE REVIEW

File: backend/internal/service/transaction_import_test.go [hunk 1]
```diff
@@ -84,6 +84,61 @@ func TestTrigramSimilarity(t *testing.T) {
...
+	cases := []struct {
+		name string
+		a    string
+		b    string
+		want bool
+	}{
+		// Issue #160 examples that must now be flagged as possible duplicates.
+		{"accent vs no accent", "Farmácia São João", "Farmacia Sao Joao", true},
+		{"cedilla folded", "Promoção", "Promocao", true},
+		{"shared word amazon", "Amazon (fraldas Luca)", "Amazon", true},
```
Comment (mateusdeitos) [inline backend/internal/service/transaction_import_test.go]: acho que esse deveria ser false, porque 'Amazon' é algo muito comum, talvez o shareSignificantWord deveria levar em conta também se as descrições possuem mais de uma palavra para considerar true, senao teremos muitos falsos positivos.

Comment (mateusdeitos) [reply backend/internal/service/transaction_import_test.go]: Boa observação. Ajustado em 4216488: o sinal de palavra em comum (`sharesSignificantWord`) agora só dispara quando **ambas** as descrições têm mais de uma palavra, então `Amazon (fraldas Luca)` x `Amazon` deixa de dar match. Os exemplos `Shopee*Drogaria` e `Zp *Sunrize` continuam funcionando (ambos os lados têm 2+ palavras), e duas descrições `Amazon` idênticas ainda casam pela similaridade de trigramas. O teste foi atualizado para `false`.

---
_Generated by [Claude Code](https://claude.ai/code)_
